### ISSUE TITLE
Since 1.3.1 - Opening screen in builder not possible

### ISSUE THREAD

**Issue by pandel** [Budibase/budibase#7630]
Hi!

Since upgrade to version 1.3.1 I am unable to open a screen in the app builder. Everything is loading "around" the page I am trying to design, but the page preview in the middle of the screen itself only shows a turning circle.

The container log shows this line (I anonymized server name and IP):

`bbproxy                                | 10.90.XX.XX - - [05/Sep/2022:09:58:37 +0000] "GET /preview HTTP/1.1" 403 265 "https://bb.machine.local:10000/builder/app/app_dev_263fbecda52a4d5baa3bceea3170394a/design/screen_ed58389259874ea8ade7c4e363591582/screens/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.5112.81 Safari/537.36 Edg/104.0.1293.47"`

I am using budibase behind a corporate proxy, which was working fine up to now. What I don't understand is the 403 as error code.

To solve this, I tried to define additional no_proxy environment vars for the bbproxy container, but that did not solve the problem.

What can I do? Or is it possible to downgrade budibase below 1.3.1?

Regards,
Holger

**Comment by pandel** [comment]
Additional info:
I reverted the container version in docker-compose.yml back to v1.2.58 and now page preview is working again.

**Comment by aptkingston** [comment]
Glad you got sorted @pandel, but feel free to raise another issue if something happens again.

**Comment by pandel** [comment]
@aptkingston No, I did not solve the main problem with 1.3.1 out, I only reverted to 1.2.58. There must be some other problem still in there...

**Comment by geertvanheusden** [comment]
I am facing the same issue after upgrading to the standalone docker image version 1.3.11.

<img width="897" alt="CleanShot 2022-09-07 at 10 13 09@2x" src="https://user-images.githubusercontent.com/1923423/188826188-a729992c-2bcc-4f62-beb6-eba5083e7f46.png">

**Comment by PGLange** [comment]
Jeah, same with my apps,
it states also 403 https://..../preview

**Comment by mike12345567** [comment]
Hi everyone, if anyone could provide an export of an app which is exhibiting this behaviour - that would help us diagnose this.

**Comment by aptkingston** [comment]
Are you all self hosting with the single docker image? Is anyone not using that?

**Comment by pandel** [comment]
> Hi everyone, if anyone could provide an export of an app which is exhibiting this behaviour - that would help us diagnose this.

I sent you an example.

**Comment by pandel** [comment]
> Are you all self hosting with the single docker image? Is anyone not using that?

@aptkingston 
Not using the single image, but the multi-image one.

**Comment by aptkingston** [comment]
@pandel could you verify you're definitely running the most up-to-date proxy image? Could you try manually pulling just to make sure - `docker pull budibase/proxy:latest`

**Comment by pandel** [comment]
@aptkingston 
As I downgraded everything to 1.2.58 to get it working somehow again, I have to take a snapshot, pull the images again and try it again. BTW, I always upgrade via docker pull, I am not using the cli command at all. So, I am going to try this and let you know, what happened.

**Comment by aptkingston** [comment]
We have a PR up to change the preview endpoint so that it does not require a new proxy config - so once that gets deployed you should be able to upgrade again and the problem will be solved.

I was just curious what the current problem is, but this PR will solve it anyway so don't worry about debugging it.
https://github.com/Budibase/budibase/pull/7670

**Comment by pandel** [comment]
@aptkingston 
Ok, then I wait. Just tried it with the last images, but the problem stays the same, even with the latest images.

**Comment by aptkingston** [comment]
Are you using a custom nginx config by any chance? Or have you mounted a volume for the proxy-service container?

**Comment by pandel** [comment]
@aptkingston 
AH! Maybe that's it. Did not think about it. Yes, I have a custom nginx conf, because I had to enable SSL. I will check the configuration file at once to see if there are any changes I do have to copy!

I never thought that this could be a proxy conf problem, sorry! I will check as fast as I can!

**Comment by aptkingston** [comment]
@pandel the change you need to add is:
```
location /preview {
  proxy_pass http://$apps:4002;
}
```

**Comment by pandel** [comment]
@aptkingston 
MANY THANKS! THAT did the trick! So, note to myself, always check nginx conf after upgrade...

**Comment by geertvanheusden** [comment]
> Are you all self hosting with the single docker image? Is anyone not using that?

@aptkingston Yes, but based on the conversation I will wait for a new release with the updated nginx file in the single docker image. 👌🏻

**Comment by aptkingston** [comment]
Hey @geertvanheusden. Yea the single docker image had a different issue - it just wasn't updated with this new endpoint. It'll be fixed in the next release :+1:

**Comment by PGLange** [comment]
@aptkingston 
I'm using Caddy to provide SSL,
and a docker-compose setup. Do I habe to place the
location /preview {
  proxy_pass http://$apps:4002;
}
inside the NGINX Container?

**Comment by aptkingston** [comment]
Hey @PGLange. Are you using Caddy instead of our proxy image, or just in front of it to provide SSL? If you're just running Caddy in front of our proxy image then you don't need to add anything.

The only people who need to make a manual change are those who are using a custom nginx config mounted inside our proxy image (which is not something we recommend doing).

**Comment by pandel** [comment]
> ... (which is not something we recommend doing).
@aptkingston ... then the easiest way, at least for having SSL support which is essential, would be to provide the possibility to choose SSL/non-SSL via environment variables by default. I don't think it should be necessary, especially in smaller setups, to provide a rev proxy for your rev proxy, only to have SSL support...

**Comment by aptkingston** [comment]
@pandel for smaller setups I would recommend using the single docker image (https://docs.budibase.com/docs/docker), which includes automatic HTTPS certificate generation for a domain of your choice via the `CUSTOM_DOMAIN` environment variable. The compose stack (or kubernetes of course) is more suited towards scalable, production deployments, where another layer of reverse proxy to handle TLS termination would be expected.

**Comment by pandel** [comment]
@aptkingston 
To better explain why...

My setup is smaller in terms of the number of involved machines, so I don't need kubernetes for a better management, ok, but it is not small in terms of complexity. But I don't need an extra reverse proxy layer, which would make everything even more complex.

The first time I went to the docs I read about the possibilities to self-host Budibase. As I drive everything in our environment via portainer, a self-hosted registry via swarm-based docker-compose stacks, I went straight to the description "Setting up Budibase" on your site ([docker-compose](https://docs.budibase.com/docs/docker-compose)). There was no hint, that one would recommend using the single image (only for ARM of course) and I don't like those monolithic containers either. So it was a no-brainer for me which solution to choose. And automatic certificate generation is useless for me, as, for some special aspects, we depend on the CA generation capabilities that our data center provides. And so we do regarding DNS handling. If you wish, it is like: I ask my data center to provide me with a new machine name and certificate and the rest is up to me.

I think, if both Budibase variants provide a web server (aka reverse proxy) to access the main application components, it should be up to the user/admin if he'd like to choose a monolithic approach or a stacked one, regardless of the complexity of the environment it is running in.

Just my2c...
